web/source/kmwcallback.ts

```diff
@@ -22,24 +22,32 @@ type KeyboardStore = PlainKeyboardStore; // | ComplexKeyboardStore;
 type RuleChar = string;
 
 class RuleDeadkey {
```

Comment:
should these be classes or interfaces; from a structural point of view should they all inherit from a base interface `RuleComponent`?

Reply:
I mean, we can write that in TypeScript, but since these correspond to incoming objects without the prototype information, I felt it better to have them merely be what they actually are - raw type specifications.